hpcpy/client.py

```diff
@@ -331,16 +331,20 @@ def submit(
         # Add queue
         if queue:
             directives.append(f"-q {queue}")
+            context["queue"] = queue
 
         # Add walltime
         if walltime:
             _walltime = str(walltime)
             directives.append(f"-l walltime={_walltime}")
+            context["walltime"] = _walltime
```

Comment:
The `str` version is more reliable, as it can be an HH:MM:SS directive OR a numerical value of seconds.